test/global_resource_test.cpp

```diff
@@ -27,25 +27,17 @@ std::size_t allocation_count = 0;
 #pragma warning (disable : 4290)
 #endif
 
-#if defined(BOOST_GCC) && (BOOST_GCC >= 40700) && (__cplusplus >= 201103L)
-#define BOOST_CONTAINER_NEW_EXCEPTION_SPECIFIER
-#define BOOST_CONTAINER_DELETE_EXCEPTION_SPECIFIER noexcept
-#else
-#define BOOST_CONTAINER_NEW_EXCEPTION_SPECIFIER    throw(std::bad_alloc)
-#define BOOST_CONTAINER_DELETE_EXCEPTION_SPECIFIER throw()
-#endif
-
 #if defined(BOOST_GCC) && (BOOST_GCC >= 50000)
 #pragma GCC diagnostic ignored "-Wsized-deallocation"
 #endif
 
-void* operator new[](std::size_t count) BOOST_CONTAINER_NEW_EXCEPTION_SPECIFIER
+void* operator new[](std::size_t count) BOOST_NOEXCEPT_IF(false)
```

Comment:
We need to support older containers that don't have noexcept and avoid warnings that tell us that new should throw std::bad_alloc. What is the problem with current code?

Reply:
Hm. I do not really understand why this commit is displayed here. To fix the splice it has nothing to do.  I did this local fix in my repository only to fix the build. But if you look at the Travis CI build (8.2), you can see next error:

libs/container/test/global_resource_test.cpp:42:41: error: ISO C++17 does not allow dynamic exception specifications [-Wdynamic-exception-spec]

void* operator new[](std::size_t count) BOOST_CONTAINER_NEW_EXCEPTION_SPECIFIER

libs/container/test/global_resource_test.cpp:34:52: note: expanded from macro 'BOOST_CONTAINER_NEW_EXCEPTION_SPECIFIER'

#define BOOST_CONTAINER_NEW_EXCEPTION_SPECIFIER    throw(std::bad_alloc)

libs/container/test/global_resource_test.cpp:42:41: note: use 'noexcept(false)' instead

Reply:
Ok, can you please open a new issue for this error, adding as much context as you can (compiler, version, flags, etc.) . I'll close this issue as your main commit was already merged. Many thanks for the report and the patch.